src/NHibernate/Mapping/ByCode/ModelMapper.cs

```diff
@@ -1327,19 +1327,19 @@ private IMapKeyRelationMapper DetermineMapKeyRelationType(MemberInfo member, Pro
 
 		private class ComponentRelationMapper : ICollectionElementRelationMapper
 		{
-			private readonly MemberInfo collectionMember;
+			private readonly PropertyPath propertyPath;
 			private readonly System.Type componentType;
 			private readonly ICustomizersHolder customizersHolder;
 			private readonly IModelInspector domainInspector;
 			private readonly ICandidatePersistentMembersProvider membersProvider;
 			private readonly ModelMapper modelMapper;
 			private readonly System.Type ownerType;
 
-			public ComponentRelationMapper(MemberInfo collectionMember, System.Type ownerType, System.Type componentType, ICandidatePersistentMembersProvider membersProvider,
+			public ComponentRelationMapper(PropertyPath propertyPath, System.Type componentType, ICandidatePersistentMembersProvider membersProvider,
 			                               IModelInspector domainInspector, ICustomizersHolder customizersHolder, ModelMapper modelMapper)
 			{
-				this.collectionMember = collectionMember;
-				this.ownerType = ownerType;
+				this.propertyPath = propertyPath;
+				this.ownerType = propertyPath.LocalMember.DeclaringType;
```

Comment:
I have found a scenario where `propertyPath.DeclaringMember.OwnerType` and `ownerType` is different: when you are mapping a member that's inherited from a base class.

I tried to write a test to reveal a defect that comes about when they are different, and I was successful. Unfortunately, this newly failing test didn't get fixed by passing the `ownerType` to `ComponentRelationMapper`.

I've pushed the failing test to this pull request - obviously I should come up with a fix now.